index.js

```diff
@@ -0,0 +1,184 @@
+const q = require('daskeyboard-applet');
+const request = require('request-promise');
+const fs = require('fs');
+const logger = q.logger;
+
+var airports = null;
+
+// Check the price of a selected flight every minute in order to buy it at the best price
+// Compare the current price to the old price stored one minute ago
+
+class FlightPriceWatcher extends q.DesktopApp {
+
+	constructor() {
+		super();
+		// every minute
+		this.pollingInterval = 60 * 1000; // ms
+		logger.info("FlightPriceWatcher starting: Get the cheapest flight price!");
+	}
+
+	async getPrice() {
+		// Get the current price of the selected flight 
+		logger.info(`Getting price`);
+		const API_BASE_URL = `https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices`;
+		const API_WORLD_URL = `browsequotes/v1.0/US/${this.config.currency}/en-US`;
+		const place = `${this.config.originPlace}-sky/${this.config.destinationPlace}-sky`;
+		const date = `${this.config.departDate}?inboundpartialdate=${this.config.returnDate}`;
+		const settings = {
+			url: `${API_BASE_URL}/${API_WORLD_URL}/${place}/${date}`,
+			method: 'GET',
+			headers: {
+				'x-rapidapi-host': 'skyscanner-skyscanner-flight-search-v1.p.rapidapi.com',
+				'x-rapidapi-key': this.authorization.apiKey
+			}
+		}
+		if (!this.authorization.apiKey) {
+			throw 'Invalid API key';
+		}
+		return request(settings).then(answer => {
+			const json = JSON.parse(answer);
+			if (json.Quotes.length == 0) {
+				return null;
+			}
+			// Collects the first element of "Quotes" corresponding to the minprice for the selected flight.
+			return json.Quotes[0].MinPrice;
+		});
+	}
+
+	// Complete the place field with a search method linked to JSON file 
+	// The JSON file holds the IATA code and the name of the airports
+	async options(fieldId, search) {
+		const API_BASE_URL = `https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices`;
+		if (airports) {
+			logger.info("Sending preloaded airports");
+			return this.getIATA(airports, search);
+		} else if (fs.existsSync('./airports.json')) {
+			logger.info('Loading places from a file.');
+			const airports = require('./airports.json');
+			return this.getIATA(airports, search);
+		} else {
+			logger.info("Retrieving airports via API...");
+			return request.get({
+				url: `${API_BASE_URL}/autosuggest/v1.0/US/${this.config.currency}/en-US/?query=${this.config.originPlace}`,
+				headers: {
+					'x-rapidapi-host': 'skyscanner-skyscanner-flight-search-v1.p.rapidapi.com',
+					'x-rapidapi-key': this.authorization.apiKey
+				},
+				json: true
+			}).then(body => {
+				airports = body;
+				return this.getIATA(airports, search);
+			}).catch((error) => {
+				logger.error("Caught error:", error);
+			})
+		}
```

Comment:
Opposite. Should try the API first and if there is an error should fallback with the local .json file